lib/internal/process/next_tick.js

```diff
@@ -5,6 +5,7 @@ exports.setup = setupNextTick;
 function setupNextTick() {
   const promises = require('internal/process/promises');
   const emitPendingUnhandledRejections = promises.setup(scheduleMicrotasks);
+  const MAX_TICKS = 1e4;
```

Comment:
@bnoordhuis or @trevnorris for more specifics on the exact semantics. Whichever it is, a comment is good